File: Solutions/1750/1750.c
/***************************************************************************************************
 * Copyright: washing
 * FileName: 1750.c
 * Author: washing
 * Version: 1.0
 * Date: 2022-12-28 10:10:35
***************************************************************************************************/

int minimumLength(char * s){
    int n = strlen(s);
    int l = 0, r = n - 1;
    while (1) {
        if (l==r || s[l] != s[r]) break;
        char c = s[l];
        while (l < n && s[l] == c) l++;
        while (r > 0 && s[r] == c) r--;
        if (r < l) {
            return 0;
        }
    }
    return r-l+1;
}

File: Solutions/1750/1750.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2022/12/28 10:03
# File: 1750.py
# Desc: 

class Solution:
    def minimumLength(self, s: str) -> int:
        while len(s) >= 2 and s[0] == s[-1]:
            s = s.strip(s[0])
        return len(s)
